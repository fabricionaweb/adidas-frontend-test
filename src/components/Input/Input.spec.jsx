import React from 'react';
import { render } from '@testing-library/react';
import * as Input from '.';

describe('Button.jsx', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Input.Button label="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Checkbox.jsx', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Input.Checkbox id="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('RadioGroup.jsx', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Input.RadioGroup values={['foo', 'bar']} id="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Text.jsx', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Input.Text id="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
