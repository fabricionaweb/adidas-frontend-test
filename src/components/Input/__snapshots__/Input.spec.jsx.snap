// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.jsx should render without crashing 1`] = `
<DocumentFragment>
  <button
    class="button"
    data-testid="button-null"
    type="button"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`Checkbox.jsx should render without crashing 1`] = `
<DocumentFragment>
  <label
    class="wrapper"
    for="test"
  >
    <input
      class="input"
      data-testid="checkbox-test"
      id="test"
      name="test"
      type="checkbox"
    />
    <span
      class="label"
    >
       
    </span>
  </label>
</DocumentFragment>
`;

exports[`RadioGroup.jsx should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <ul
      class="list"
    >
      <li>
        <label
          for="test_0"
        >
          <input
            class="input"
            id="test_0"
            name="test"
            type="radio"
            value="foo"
          />
          <span
            class="label"
          >
            foo
          </span>
        </label>
      </li>
      <li>
        <label
          for="test_1"
        >
          <input
            class="input"
            id="test_1"
            name="test"
            type="radio"
            value="bar"
          />
          <span
            class="label"
          >
            bar
          </span>
        </label>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Text.jsx should render without crashing 1`] = `
<DocumentFragment>
  <label
    class="label"
    for="test"
  >
     
    <input
      class="input"
      data-testid="input-test"
      id="test"
      name="test"
      type="text"
    />
  </label>
</DocumentFragment>
`;
